packages/dashboard/src/components/sidePanel/sections/baseSettingSection/index.tsx

```diff
@@ -89,28 +106,47 @@ export const BaseSettings: FC<Widget> = (widget) => {
         />
 
         <label htmlFor='base-settings-width' className='section-item-label'>
-          {defaultMessages.width}
+          Width
         </label>
-        <Input
-          controlId='base-settings-width'
-          value={`${formattedValue.width}`}
-          type='number'
-          onChange={onWidthChange}
-          data-test-id='base-setting-width-input'
+
+        <Controller
+          name='width'
+          control={control}
+          render={({ field: { value } }) => (
+            <Input
+              controlId='base-settings-width'
+              data-test-id='base-setting-width-input'
+              invalid={Boolean(errors.width)}
+              onBlur={() => trigger('width')}
```

Comment:
Blocker: Use [`useForm({ mode: 'onBlur' })`](https://react-hook-form.com/api/useform/#mode) instead of manually triggering first error.